stacks-signer/src/signer.rs

```diff
@@ -0,0 +1,2005 @@
+                if let Err(e) = self
+                    .stackerdb
+                    .send_message_with_retry(block_validate_reject.clone().into())
+                {
+                    warn!(
+                        "Signer #{}: Failed to send block rejection to stacker-db: {e:?}",
+                        self.signer_id
+                    );
+                }
+                block_info
+            }
+        };
+        if let Some(mut nonce_request) = block_info.nonce_request.take() {
+            debug!("Signer #{}: Received a block validate response from the stacks node for a block we already received a nonce request for. Responding to the nonce request...", self.signer_id);
+            // We have received validation from the stacks node. Determine our vote and update the request message
+            Self::determine_vote(self.signer_id, block_info, &mut nonce_request);
+            // Send the nonce request through with our vote
+            let packet = Packet {
+                msg: Message::NonceRequest(nonce_request),
+                sig: vec![],
+            };
+            self.handle_packets(stacks_client, res, &[packet], current_reward_cycle);
+        } else {
+            let coordinator_id = self.coordinator_selector.get_coordinator().0;
+            if block_info.valid.unwrap_or(false)
+                && !block_info.signed_over
+                && coordinator_id == self.signer_id
+            {
+                // We are the coordinator. Trigger a signing round for this block
+                debug!(
+                    "Signer #{}: triggering a signing round over the block {}",
+                    self.signer_id,
+                    block_info.block.header.block_hash()
+                );
+                self.commands.push_back(Command::Sign {
+                    block: block_info.block.clone(),
+                    is_taproot: false,
+                    merkle_root: None,
+                });
+            } else {
+                debug!(
+                    "Signer #{} ignoring block.", self.signer_id;
+                    "block_hash" => block_info.block.header.block_hash(),
+                    "valid" => block_info.valid,
+                    "signed_over" => block_info.signed_over,
+                    "coordinator_id" => coordinator_id,
+                );
+            }
+        }
+    }
+
+    /// Handle signer messages submitted to signers stackerdb
+    fn handle_signer_messages(
+        &mut self,
+        stacks_client: &StacksClient,
+        res: Sender<Vec<OperationResult>>,
+        messages: &[SignerMessage],
+        current_reward_cycle: u64,
+    ) {
+        let coordinator_pubkey = self.coordinator_selector.get_coordinator().1;
+        let packets: Vec<Packet> = messages
+            .iter()
+            .filter_map(|msg| match msg {
+                // TODO: should we store the received transactions on the side and use them rather than directly querying the stacker db slots?
```

Comment:
Probably not necessary for now, since the signer has a dedicated stacks node that can just serve this on demand.